File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int Ways(int n, vector<int>& Steps) {
        if (n < 0)
            return 0;
        else if (n == 0)
            return 1;
        if (Steps[n - 1] != 0)
            return Steps[n - 1];

        Steps[n - 1] = Ways(n - 1, Steps) + Ways(n - 2, Steps);
        return Steps[n - 1];
    }
    int climbStairs(int n) {
        std::vector<int> Steps(n, 0);
        return Ways(n, Steps);
    }
};

File: 70-climbing-stairs/climbing-stairs.py
import numpy as np
class Solution:
    # Memoization Technique
    def CountingStairs(self,n,array):
        if(n==0):
            return 1
        elif(n<0):
            return 0
        elif(array[n-1]!=0):
            return array[n-1]
        array[n-1]=self.CountingStairs(n-1,array)+self.CountingStairs(n-2,array)
        return array[n-1]
    def climbStairs(self, n: int) -> int:
        if(n<0):
            return 0
        array=np.zeros(n)
        array=self.CountingStairs(n,array)
        return int(array)


        